c/dynalloc.c

```diff
@@ -1164,6 +1165,274 @@ int dynallocUnallocDatasetByDDName(const DynallocDDName *ddName,
+  if (request->recordFormat != NULL) {
+    snprintf(field, sizeof(field)-1, "RECFM(%s) ", request->recordFormat);
+    strncat(buffer, field, bufferSize - strlen(buffer));
+  }
+
+  /* Initialize record length */
+  if (request->recordLength != 0) {
+    snprintf(field, sizeof(field)-1, "LRECL(%d) ", request->recordLength);
+    strncat(buffer, field, bufferSize - strlen(buffer));
+  }
+
+  /* Initialize block size*/
+  if (request->blkSize != 0) {
+    snprintf(field, sizeof(field)-1, "BLKSIZE(%d) ", request->blkSize);
+    strncat(buffer, field, bufferSize - strlen(buffer));
+  }
+
+  /* Initialize storage type */
+  if (request->storageClass != NULL) {
+    snprintf(field, sizeof(field)-1, "STORCLAS(%s) ", request->storageClass);
+    strncat(buffer, field, bufferSize - strlen(buffer));
+  }
+
+  /* Initialize  data type */
+  if (request->fileData != NULL) {
+    snprintf(field, sizeof(field)-1, "FILEDATA(%s) ", request->fileData);
+    strncat(buffer, field, bufferSize - strlen(buffer));
+  }
+
+  /* Initialize extended attribute */
+  if (request->eattr != NULL) {
+    strncpy(tempString, request->eattr, sizeof (tempString) -1);
+    strupcase(tempString);
+    if (strncmp(tempString, "TRUE", 5)) {
+      snprintf(field, sizeof(field)-1, "EATTR(OPT) ");
+      strncat(buffer, field, bufferSize - strlen(buffer));
+    }
+  }
+
+  /* Initialize number of generations */
+  if (request->numGenerations != 0) {
+    snprintf(field, sizeof(field)-1, "MAXGENS(%d) ", request->numGenerations);
+    strncat(buffer, field, bufferSize - strlen(buffer));
+  }
+
+  /* Initialize volumn if defined */
+  if (request->volume != NULL) {
+    snprintf(field, sizeof(field)-1, "VOL(%s) ", request->volume);
+    strncat(buffer, field, bufferSize - strlen(buffer));
+  }
+
+  /* Initialize dataClass if defined */
+  if (request->dataClass != NULL) {
+    snprintf(field, sizeof(field)-1, "DATACLAS(%s) ", request->dataClass);
+    strncat(buffer, field, bufferSize - strlen(buffer));
+  }
+
+  /* Initialize management class if defined */
+  if (request->manageClass != NULL) {
+    snprintf(field, sizeof(field)-1, "MGMTCLAS(%s) ", request->manageClass);
+    strncat(buffer, field, bufferSize - strlen(buffer));
+  }
+
+  /* Initialize management class if defined */
+  if (request->averageRecord != NULL) {
+    snprintf(field, sizeof(field)-1, "AVGREC(%s) ", request->averageRecord);
+    strncat(buffer, field, bufferSize - strlen(buffer));
+  }
+
+  /* Initialize expiration data  */
+  if (request->expiration != NULL) {
+    snprintf(field, sizeof(field)-1, "EXPDL(%s) ", request->expiration);
+    strncat(buffer, field, bufferSize - strlen(buffer));
+  }
+
+  /* Initialize expiration data  */
+  if (request->spaceUnit != NULL) {
+    snprintf(field, sizeof(field)-1, "SPACE(%s) ", request->spaceUnit);
+    strncat(buffer, field, bufferSize - strlen(buffer));
+  }
+  return status;
+}
+
+static void  allocDataSetCloseString(dataSetRequest *request, char *buffer,
```

Comment:
What about reusing `dynallocUnallocDatasetByDDName` and the other unalloc functions?